mcs/class/System.Runtime.Remoting/System.Runtime.Remoting.Channels.Tcp/TcpServerChannel.cs

```diff
@@ -259,6 +264,22 @@ public void StopListening (object data)
 			server_thread.Join ();
 			server_thread = null;			
 		}
+
+		private static IPAddress GetMachineAddress (IPHostEntry host, AddressFamily addressFamily)
```

Comment:
This is taken mostly verbatim from: http://referencesource.microsoft.com/#System.Runtime.Remoting/channels/core/corechannel.cs,1dc73914f7b05ec4